test/test.js

```diff
@@ -150,6 +150,198 @@ var util = require('util')
       })
     });
 
+    describe('shouldSetVaryHeader', function() {
+      it('Vary: "Access-Control-Request-Headers" header is set by default for pre-flight requests', function (done) {
+        var cb = after(1, done)
+
+        var req = new FakeRequest('OPTIONS')
+        var res = new FakeResponse()
+        req.host = 'http://example.com'
+        req.originalUrl = '/images/asdf.png'
+
+
+        res.on('finish', function () {
+          assert.equal(res.statusCode, 204)
+          assert.equal(res.getHeader('Vary'), 'Access-Control-Request-Headers')
+          cb()
+        })
+
+        cors()(req, res, function (err) {
+          cb(err || new Error('should not be called'))
+        })
+      })
+
+      it('Vary: "Origin, Access-Control-Request-Headers" header is set by default for pre-flight requests with origins', function (done) {
+        var cb = after(1, done)
+
+        var req = new FakeRequest('OPTIONS')
+        var res = new FakeResponse()
+        req.host = 'http://example.com'
+        req.originalUrl = '/images/asdf.png'
+        var options = {
+          origin: 'https://example.com'
+        }
+
+        res.on('finish', function () {
+          assert.equal(res.statusCode, 204)
+          assert.equal(res.getHeader('Vary'), 'Origin, Access-Control-Request-Headers')
+          cb()
+        })
+
+        cors(options)(req, res, function (err) {
+          cb(err || new Error('should not be called'))
+        })
+      })
+
+      it('Vary: "Access-Control-Request-Headers" header is set for pre-flight requests and shouldSetVaryHeader(req, \'Origin\') configured to return false with origins', function (done) {
+        var cb = after(1, done)
+
+        var req = new FakeRequest('OPTIONS')
+        var res = new FakeResponse()
+        req.host = 'http://example.com'
+        req.originalUrl = '/images/asdf.png'
+        var options = {
+          origin: 'https://example.com',
+          shouldSetVaryHeader: function(req, header) {
+            if (header === 'Access-Control-Request-Headers') {
+              return true
+            }
+
+            if (req.originalUrl.startsWith('/images') && header === 'Origin') {
```

Comment:
Node `0.10` failures are due to lack of `.startsWith` this language feature came in with ES6 in Node 6

Reply:
`if (req.originalUrl.indexOf('/images') === 0 && header === 'Origin') {`

indexOf should work.

Reply:
Made this change!